File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 1: return True
        dp = [False for i in range(len(nums))]
        dp[-1] = True
        for i in range(len(nums) - 2, -1, -1):
            if nums[i] + i >= len(nums) - 1: dp[i] = True
            else:
                for j in range(i + 1, i + nums[i] + 1):
                    if dp[j]:
                        dp[i] = True
                        break
        return dp[0]

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        if (nums.length == 1) {
            return true;
        }

        boolean[] dp = new boolean[nums.length];
        dp[nums.length - 1] = true;

        for (int i = nums.length - 2; i > -1; i--) {
            if (nums[i] + i >= nums.length - 1) {
                dp[i] = true;
            } else {
                for (int j = i + 1; j < i + nums[i] + 1; j++) {
                    if (dp[j]) {
                        dp[i] = true;
                        break;
                    }
                }
            }
        }
        return dp[0];
    }
}